ext/afform/core/ang/af/afForm.component.js

```diff
@@ -90,11 +90,13 @@
           ctrl.showSubmitButton = displaySubmitButton(args);
         }
         if (toLoad) {
+          $element.block();
```

Comment:
@ufundo currently this will block the form during all prefill operations, including the ones that happen when you select one entity from a dropdown. That might be a bit much... at least for a start maybe we should just do the whole-form prefills & move this a couple lines up into the `params.fillMode = 'entity'` condition above.

Reply:
I was trying to do that but I couldn't quite get my head around the possible codepaths, and I wanted to keep the block as close to the start of the async call as possible (and consequently as close to the unblock as possible).

Reply:
@ufundo it's just a conditional for whether the whole form or just a specific part of the form is being prefilled. If you move the block() up a couple lines into the whole-form conditional, then we should be good.

Reply:
I was worried that `toLoad` might be false. I think that's not possible as it stands, but it feels a bit brittle.

how about

```suggestion
          if (params.fillMode === 'form') {
            $element.block();
          }
```

Reply:
Yea that sounds good.